resources/views/components/input/select.blade.php

```diff
@@ -1,33 +1,73 @@
+<?php
+
+use Illuminate\Database\Eloquent\Model;
+use Illuminate\Support\Collection;
+use Illuminate\Support\Str;
+?>
+
+@props([
+    'disabled' => false,
+    'fullWidth' => true,
+    'help' => null,
+    'icon' => false,
+    'id' => null,
+    'inline' => false,
+    'label' => true,
+    'options' => collect(),
+    'model' => null,
+    'name' => null,
+    'placeholder' => false,
+    'readonly' => false,
+    'required' => false,
+    'requiredSilent' => false,
+    'value' => null,
+])
+
+<?php
+if ($model && !$model instanceof Model) {
+    throw new Exception('"model" property is not an Eloquent model');
+}
+
+$options = !$options instanceof Collection ? collect($options) : $options;
+
+$id = $id ?: 'input_' . Str::random();
+?>
+
 <x-form-field
     :model="$model ?? null"
-    :attribute="$attribute"
+    :name="$name"
     :help="$help ?? null"
 >
-    <x-slot name="label">
-        @if($errors->has($attribute))
-            <x-far-times-circle /> {{ $errors->first($attribute) }}
-        @else
-            {{ __('validation.attributes.'.$attribute) }} {{ !empty($required) ? '*' : '' }}
-        @endif
-    </x-slot>
-    @if($options->count() == 1 && !empty($required))
-        <?php
-        $value = $options->keys()->first();
-        $label = $options->first();
-        ?>
-        <div class="form-control form-control-static">
-            {{ $label }}
-        </div>
-        <input type="hidden" id="{{ $attribute }}" name="{{ $attribute }}" value="{{ $value }}">
-    @elseif($options->count() > 1 || empty($required))
-        <select class="form-control" id="{{ $attribute }}" name="{{ $attribute }}">
+    @if($label)
+        <x-slot name="label">
+            {{ $label === true ? __('validation.attributes.' . strtolower($name)) : $label }} {{ $required && !$requiredSilent ? '*' : '' }}
+        </x-slot>
+    @endif
```

Comment:
Similar to other feedback, `$label` should likely always be present.

Reply:
Addressed in https://github.com/RetroAchievements/RAWeb/pull/2247/commits/7aaaa88ce62aa676d6a2b56cbf78db0a530a4a37